p2p: Fix flaky TestSwitchReconnectsToPersistentPeer

Our CI runs keep failing non-determinstically in the p2p package, we should remove the flakiness. The specific error is:

```
ok  	github.com/tendermint/tendermint/lite/files	1.644s	coverage: 87.0% of statements
--- FAIL: TestSwitchReconnectsToPersistentPeer (0.03s)
	Error Trace:	switch_test.go:282
	Error:      	Expected value not to be nil.
	Test:       	TestSwitchReconnectsToPersistentPeer
FAIL
```

--- a/p2p/peer_test.go
+++ b/p2p/peer_test.go
@@ -140,6 +140,8 @@ func (p *remotePeer) Stop() {
 }
 
 func (p *remotePeer) accept(l net.Listener) {
+	conns := []net.Conn{}
+
 	for {
 		conn, err := l.Accept()
 		if err != nil {
@@ -160,10 +162,15 @@ func (p *remotePeer) accept(l net.Listener) {
 		if err != nil {
 			golog.Fatalf("Failed to perform handshake: %+v", err)
 		}
+
+		conns = append(conns, conn)
+
 		select {
 		case <-p.quit:
-			if err := conn.Close(); err != nil {
-				golog.Fatal(err)
+			for _, conn := range conns {
+				if err := conn.Close(); err != nil {
+					golog.Fatal(err)
+				}
 			}
 			return
 		default:

